addons/survey/models/survey_user.py

```diff
@@ -359,6 +362,47 @@ def _prepare_statistics(self):
             {'text': _("Unanswered"), 'count': res[user_input]['skipped']}
         ] for user_input in self]
 
+    # ------------------------------------------------------------
+    # Conditional Management
+    # ------------------------------------------------------------
+    def _get_selected_suggested_answers(self):
+        selected_suggested_answers_ids = []
+        for answer in self.user_input_line_ids:
+            if answer.question_id.question_type in ['simple_choice', 'multiple_choice']:
+                selected_suggested_answers_ids.append(answer.suggested_answer_id.id)
+        return selected_suggested_answers_ids
+
+    def _clear_inactive_conditional_answers(self):
```

Comment:
Nope, because the is important for scoring. If we keep answers that should not be answered, scoring will be fucked up, Also, if the selected answer triggers another question later in the survey (here this method is used mainly for page per question survey, we have a similar method in JS for the other survey layout), if the answer is not cleared, a question that should not be displayed to the user will be, and then kaboom !